src/crimson/os/seastore/cached_extent.h

```diff
@@ -760,9 +766,25 @@ class CachedExtent
   trans_view_set_t retired_transactions;
 
   CachedExtent(CachedExtent &&other) = delete;
-  CachedExtent(ceph::bufferptr &&_ptr) : ptr(std::move(_ptr)) {
-    length = ptr->length();
+
+  /// construct a fully loaded CachedExtent
+  CachedExtent(ceph::bufferptr &&_ptr)
+    : length(_ptr.length()) {
+    ptr = std::move(_ptr);
```

Comment:
Why not just `CachedExtent(ceph::bufferptr &&_ptr) : ptr(std::move(_ptr), length(ptr->length()) {}`?

Reply:
I think this way is less error-prone, which doesn't depend on the initialization order.